tests/chat_test.py

```diff
@@ -0,0 +1,76 @@
+import asyncio
+import os
+import shutil
+import subprocess
+import tempfile
+import time
+
+from pykeybasebot import Bot, kbsubmit
+from pykeybasebot.types import chat1
+
+import pytest
+import yaml
+
+
+# create temp dir
+def create_working_dir():
+    with tempfile.TemporaryDirectory() as tmpdir:
+        kb_location = shutil.which("keybase").strip()
+        kb_destination = f"{tmpdir}/keybase"
+        shutil.copy(kb_location, kb_destination)
+        print(os.listdir(tmpdir))
+
+
+@pytest.fixture()
+def config():
+    with open("tests/test_config.yaml") as f:
+        data = yaml.safe_load(f)
+        return data
+
+
+@pytest.fixture()
+def channel(config):
+    return chat1.ChatChannel(
+        name=f"{config['bots']['alice']['username']},{config['bots']['bob']['username']}"
+    )
+
+
+@pytest.fixture()
+async def bot(config):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        kb_location = shutil.which("keybase").strip()
+        kb_destination = f"{tmpdir}/keybase"
+        shutil.copy(kb_location, kb_destination)
+
+        args = [kb_destination, "--home", tmpdir, "service"]
+        subprocess.Popen(args)
+        time.sleep(2)
+
+        bot = Bot(
+            username=config["bots"]["alice"]["username"],
```

Comment:
I'm not totally happy with how we can't specify an argument to the fixture to handle other bots: "bob", "charlie", etc. I'm pretty new to pytest fixtures, would love to hear if there's a way to do this.

Reply:
the fixture can be a function that takes an argument and returns the thing you want.

Reply:
I don't think I can also do the teardown code in the fixture then. I was playing around with something like this, but couldn't figure out scoping.
```
@pytest.fixture()
async def bot(config):
     ...
    def setup_bot(name):
        bot = Bot(
            username=config["bots"][name]["username"],
            paperkey=config["bots"][name]["paperkey"],
            handler=noop_handler,
            home_path=tmpdir,
            keybase=kb_destination,
        )
        return bot
   yield setup_bot
   
   await bot.teardown() # error here, what does `bot` refer to?
   await kbsubmit(kb_destination, "ctl stop")
```

The issue to me seems to be that the bot is logged in when it's instantiated. If we could instantiate the bot, then call `bot.init(username, paperkey)` in `setup_bot`, we could tear it down outside of `setup_bot`. 

Or maybe I'm missing something about setup/teardown in fixtures?

Reply:
does it work if you make `bot` nonlocal?